src/matcher/many_patterns/naive.rs

```diff
@@ -1,78 +1,89 @@
-use std::hash::Hash;
-
-use portgraph::{LinkView, NodeIndex};
-
 use crate::{
-    matcher::{Match, PatternMatch, PortMatcher, SinglePatternMatcher},
-    patterns::UnweightedEdge,
-    EdgeProperty, NodeProperty, Pattern, PatternID, Universe,
+    constraint::Constraint,
+    indexing::{IndexKey, IndexValue},
+    matcher::{PatternMatch, PortMatcher, SinglePatternMatcher},
+    pattern::Pattern,
+    HashMap, IndexingScheme, Predicate,
 };
 
 /// A simple matcher for matching multiple patterns.
```

Comment:
I'd try to remark that you almost never want to use this (I think?), and link to the ManyMatcher.

Reply:
Yes I'm adding that.